### ISSUE TITLE
Can't Run Tests Locally

### ISSUE THREAD

**Issue by jalcine** [getsentry/sentry-elixir#220]
Running Elixir 1.5.2 on Erlang 20 gives me this off a master build:

```shell
✔ ~/src/sentry-elixir [supporting-confex L|⚑ 1]
02:06 $ mix test


  1) test parses error exception (Sentry.EventTest)
     test/event_test.exs:14
     Assertion with == failed
     code:  assert event.stacktrace() == %{frames: Enum.reverse([%{filename: nil, function: "Sentry.Event.not_a_function/3", lineno: nil, module: Sentry.Event, context_line: nil, post_context: [], pre_context: [], in_app: false, vars: %{"arg0" => "1", "arg1" => "2", "arg2" => "3"}}, %{filename: "test/event_test.exs", function: "Sentry.EventTest.event_generated_by_exception/1", lineno: 8, module: Sentry.EventTest, context_line: nil, post_context: [], pre_context: [], in_app: false, vars: %{}}, %{filename: "test/event_test.exs", function: "Sentry.EventTest.\"test parses error exception\"/1", lineno: 15, module: Sentry.EventTest, context_line: nil, post_context: [], pre_context: [], in_app: false, vars: %{}}, %{filename: "lib/ex_unit/runner.ex", function: "ExUnit.Runner.exec_test/1", lineno: 302, module: ExUnit.Runner, context_line: nil, post_context: [], pre_context: [], in_app: false, vars: %{}}, %{filename: "timer.erl", function: ":timer.tc/1", lineno: 166, module: :timer, context_line: nil, post_context: [], pre_context: [], in_app: false, vars: %{}}, %{filename: "lib/ex_unit/runner.ex", function: "anonymous fn/3 in ExUnit.Runner.spawn_test/3", lineno: 250, module: ExUnit.Runner, context_line: nil, post_context: [], pre_context: [], in_app: false, vars: %{}}])}
     left:  %{frames: [%{context_line: nil, filename: "lib/ex_unit/runner.ex", function: "anonymous fn/3 in ExUnit.Runner.spawn_test/3", in_app: false, module: ExUnit.Runner, post_context: [], pre_context: [], vars: %{}, lineno: 240}, %{context_line: nil, filename: "timer.erl", function: ":timer.tc/1", in_app: false, lineno: 166, module: :timer, post_context: [], pre_context: [], vars: %{}}, %{context_line: nil, filename: "lib/ex_unit/runner.ex", function: "ExUnit.Runner.exec_test/1", in_app: false, module: ExUnit.Runner, post_context: [], pre_context: [], vars: %{}, lineno: 292}, %{context_line: nil, filename: "test/event_test.exs", function: "Sentry.EventTest.\"test parseserror exception\"/1", in_app: false, lineno: 15, module: Sentry.EventTest, post_context: [], pre_context: [], vars: %{}}, %{context_line: nil, filename: "test/event_test.exs", function: "Sentry.EventTest.event_generated_by_exception/1", in_app: false, lineno: 8, module: Sentry.EventTest, post_context: [], pre_context: [], vars: %{}}, %{context_line: nil, filename: nil, function: "Sentry.Event.not_a_function/3", in_app: false, lineno: nil, module: Sentry.Event, post_context: [], pre_context: [], vars: %{"arg0" => "1", "arg1" => "2", "arg2" => "3"}}]}
     right: %{frames: [%{context_line: nil, filename: "lib/ex_unit/runner.ex", function: "anonymous fn/3 in ExUnit.Runner.spawn_test/3", in_app: false, module: ExUnit.Runner, post_context: [], pre_context: [], vars: %{}, lineno: 250}, %{context_line: nil, filename: "timer.erl", function: ":timer.tc/1", in_app: false, lineno: 166, module: :timer, post_context: [], pre_context: [], vars: %{}}, %{context_line: nil, filename: "lib/ex_unit/runner.ex", function: "ExUnit.Runner.exec_test/1", in_app: false, module: ExUnit.Runner, post_context: [], pre_context: [], vars: %{}, lineno: 302}, %{context_line: nil, filename: "test/event_test.exs", function: "Sentry.EventTest.\"test parseserror exception\"/1", in_app: false, lineno: 15, module: Sentry.EventTest, post_context: [], pre_context: [], vars: %{}}, %{context_line: nil, filename: "test/event_test.exs", function: "Sentry.EventTest.event_generated_by_exception/1", in_app: false, lineno: 8, module: Sentry.EventTest, post_context: [], pre_context: [], vars: %{}}, %{context_line: nil, filename: nil, function: "Sentry.Event.not_a_function/3", in_app: false, lineno: nil, module: Sentry.Event, post_context: [], pre_context: [], vars: %{"arg0" => "1", "arg1" => "2", "arg2" => "3"}}]}
     stacktrace:
       test/event_test.exs:28: (test)

...............................

  2) test Bad function call causing GenServer crash makes call to Sentry API (Sentry.LoggerTest)
     test/logger_test.exs:81
     Assertion with == failed
     code:  assert List.first(json["exception"])["type"] == "exit"
     left:  "Elixir.FunctionClauseError"
     right: "exit"
     stacktrace:
       test/logger_test.exs:87: anonymous fn/1 in Sentry.LoggerTest."test Bad function call causing GenServer crash makes call to Sentry API"/1
       (bypass) lib/bypass/plug.ex:11: Bypass.Plug.call/2
       (plug) lib/plug/adapters/cowboy/handler.ex:15: Plug.Adapters.Cowboy.Handler.upgrade/4
       (cowboy) /home/okoye/Development/prjs/sentry-elixir/deps/cowboy/src/cowboy_protocol.erl:442: :cowboy_protocol.execute/4

.................

Finished in 0.7 seconds
50 tests, 2 failures

Randomized with seed 941598
```

Though not immediate visible in that stacktrace, I did see what the error message was for the first one:

![screenshot_20171018_020803](https://user-images.githubusercontent.com/452100/31712949-7c948124-b3b1-11e7-8c73-82b9fe1fcf44.png)
Looks like a hard check on the line numbers used in ExUnit; something shifted in 1.5.x

The second one was a bit more obscure but it looks like the underlying Erlang type for exits might have been converted into something specific for Elixir now when it comes to trapping exit codes.

**Comment by mitchellhenke** [comment]
@jalcine hey, thanks for opening this 🙂 

The first test is (one of) the tests that is hardcoded and brittle when it comes to different versions.

I have tried to make them more flexible, but important parts of stacktraces get changed between versions.  [This test](https://github.com/getsentry/sentry-elixir/blob/master/test/logger_test.exs#L90) is an example.  The method `NaiveDateTime.from_erl/2` in 1.4 does not exist in 1.5, as it now has a default argument and is `NaiveDateTime.from_erl/3`, so the test would fail.

It's tough for me to balance supporting the tests working well across versions, and having them be specific.  In the mean time, I should have some documentation around this, as it is surprising to have tests fail in this manner.

The second test is something I have not been able to investigate just yet.

**Comment by jalcine** [comment]
I have a small patch in the works at the moment that could take care of this.

**Comment by jalcine** [comment]
Just completed and cleaned up the changes in PR #221.

**Comment by mitchellhenke** [comment]
Closed by @jalcine #221
